contentcuration/search/viewsets/contentnode.py

```diff
@@ -162,7 +162,7 @@ def annotate_queryset(self, queryset):
         2. Annotate lists of content node and channel pks
         """
         # Get accessible content nodes that match the content id
-        content_id_query = ContentNode.filter_view_queryset(ContentNode.objects.all(), self.request.user).filter(
+        content_id_query = queryset.filter(
```

Comment:
Unfortunately, this will still duplicate the permissions logic in the resulting query since this will produce a subquery with the same filtering. So there's still potential for optimizing this by reducing the duplication, perhaps by utilizing a CTE.

Reply:
This is not duplicating the permissions logic I think. It is just filtering the queryset for use in `location_ids`. There's no permission check being made here.

Reply:
The code does not duplicate the logic, but the resulting query does. This is because `content_id_query` is used within a subquery in the final query expression, as you can see in `SQArrayAgg(content_id_query, field="id")`, so any filtering applied to the queryset will end up duplicated in the outer and sub queries. Given the `channel_id` is an [annotation created by a subquery](https://github.com/learningequality/studio/blob/unstable/contentcuration/contentcuration/viewsets/contentnode.py#L690), I suspect the resulting query would look something like this:
```sql
SELECT
    (
        SELECT
            ch1.id
        FROM contentcuration_channel ch1
            INNER JOIN contentcuration_contentnode cn1
                ON cn1.id = ch1.main_tree_id
        WHERE cn1.tree_id = cn2.tree_id
        LIMIT 1
    ) AS channel_id,
    ARRAY_AGG(
        SELECT
            cn3.id
        FROM contentcuration_contentnode cn3
        WHERE cn3.content_id = cn2.content_id
            AND cn3.tree_id IN (
                SELECT
                    cn4.tree_id
                FROM contentcuration_channel ch2
                    INNER JOIN contentcuration_contentnode cn4
                        ON cn4.id = ch2.main_tree_id
                WHERE ch2.id IN (<permittable channel IDs>)
            )
            AND <other search filters>
    ) AS location_ids,
    ...<other annotations and selected fields>
FROM contentcuration_contentnode cn2
WHERE channel_id IN (<permittable channel IDs>) AND <other search filters>
```
The issue we've been discussing on slack is the presence of `<other search filters>` in the subquery for `location_ids`.

Reply:
Resolved on Slack. We gonna remove the queries related to `location_id`s.